Bio/SeqIO/QualityIO.py

```diff
@@ -1031,10 +1035,13 @@ def __init__(self, source):
 
     def __next__(self) -> SeqRecord:
         """Parse the file and generate SeqRecord objects."""
+
+        # NamedTemporaryFile does not support __next__ so create an iter as a work-around
+        stream = iter(self.stream)
         line = self.line
         if line is None:
             try:
-                line = next(self.stream)
+                line = next(stream)
```

Comment:
Same here. The current fix required calling `iter` each time we enter  `__next__`.